### ISSUE TITLE
Messages from group chats are not resending and usually not fetching from offline

### ISSUE THREAD

**Issue by churik** [status-im/status-react#20247]
# Bug Report

## Problem 
Messages that sent from offline from the mobile device are not resent and usually messages that should be fetched from oflline, are not fetched.
The issue seems to be specific to group chats, as re-send is working fine for communites and 1-1 chats

### Reproduction
**Prerequisites:** users (IOS, Android, desktop) are group chat members

- IOS, Android: go  offline
- IOS, Android: send 5 messages
- Desktop: send 5 messages for fetching
- wait 5 mins
- IOS, Android: go online and wait for messages to arrive

Result:
- IOS, Android: 5 messages sent from offline are never resent (waited for 20 mins
- IOS: 5 messages that should be fetched

### Additional Information

- Status version: nightly 
- Operating System: Android, iOS

**Android:**
![IMAGE 2024-05-29 15:17:00](https://github.com/status-im/status-mobile/assets/4557972/391e0dc1-1d19-4d37-9521-1d03183be401)

**IOS:**
![IMAGE 2024-05-29 15:19:11](https://github.com/status-im/status-mobile/assets/4557972/8f226a55-9ee3-42ff-9206-7e7f9990e98e)

[comment]: # (Please, add logs/notes if necessary)
IOS:[logs (86).zip](https://github.com/status-im/status-mobile/files/15484879/logs.86.zip)

Android: [Status-debug-logs.zip](https://github.com/status-im/status-mobile/files/15484882/Status-debug-logs.zip)

**Comment by cammellos** [comment]
This is due to the fact that messages in group chats are not sent using datasync anymore, that's a recent regression, will push a fix. It also looks like they are not retried properly, so we need to investigate a bit, but might be a red herring.

**Comment by pavloburykh** [comment]
@cammellos here is another issue form @VolodLytvynenko, please take a look.

### ISSUE 2 Messages send from offline are not delivered to receiver after sender is back online in community


**Steps:**

1. User A and B open same community channel
2. User A goes offline
3. User A sends messages from offline
4. User A goes online
5. Wait until messages are recend
6. See if all messages are delivered to User B

**Actual result:** not all of messages are delivered to receiver


[andr_sender.zip](https://github.com/status-im/status-mobile/files/15499046/andr_sender.zip)
[ios_receiver_.zip](https://github.com/status-im/status-mobile/files/15499047/ios_receiver_.zip)

https://github.com/status-im/status-mobile/assets/97245802/2fb69636-498d-47fb-a1e5-45e2aa0c1c0b
